src/components/icon/icon.spec.js

```diff
@@ -514,29 +510,39 @@ describe('MdIcon service', function() {
     });
 
     describe('icon group is not found', function() {
-      it('should log Error', function() {
-        var msg;
-        try {
-          $mdIcon('notfound:someIcon')
-            .catch(function(error) {
-              msg = error.data;
-            });
+      it('should log Error and reject', inject(function($log) {
+        var errorMessage = 'Cannot GET notfoundgroup.svg';
+        var caughtRejection = false;
+
+        $mdIcon('notfound:someIcon')
+          .catch(function(error) {
+            expect(error.data).toBe(errorMessage);
+            caughtRejection = true;
+          });
 
-          $httpBackend.flush();
-        } finally {
-          expect(msg).toEqual('Cannot GET notfoundgroup.svg');
-        }
-      });
+        $httpBackend.flush();
+
+        expect(caughtRejection).toBe(true);
+        expect($log.warn.logs[0]).toEqual([errorMessage]);
+      }));
     });
 
     describe('icon is not found', function() {
-      it('should not throw Error', function() {
-        expect(function(){
-          $mdIcon('notfound');
+      it('should log Error and reject', inject(function($log) {
```

Comment:
The original behavior in 1.3, 1.4 and 1.5 was to not throw any error if the user called `$mdIcon()` with an unknown icon. Since 1.6 was going to throw a PUR, we decided to update the test to just check for the specific error (to account for 1.6) and instruct users to use `.catch()` or disable the new PUR checks if they wanted the old pre 1.6 behavior.

@ThomasBurleson Any thoughts on whether we should keep the "don't throw an error" behavior, or if we should make a breaking change to say that it will now throw an error/reject which should always be caught?